test/e2e/externalplugin/generate_test.go

```diff
@@ -0,0 +1,114 @@
+/*
+Copyright 2023 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+   http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package externalplugin
+
+import (
+	"path/filepath"
+
+	pluginutil "sigs.k8s.io/kubebuilder/v3/pkg/plugin/util"
+
+	//nolint:golint
+	//nolint:revive
+	. "github.com/onsi/ginkgo/v2"
+
+	//nolint:golint
+	//nolint:revive
+	. "github.com/onsi/gomega"
+
+	//nolint:golint
+	//nolint:revive
+	//nolint:golint
+	//nolint:revive
+	"sigs.k8s.io/kubebuilder/v3/test/e2e/utils"
+)
+
+var _ = Describe("kubebuilder", func() {
+	Context("plugin sampleexternalplugin/v1", func() {
+		var (
+			kbc *utils.TestContext
+		)
+
+		BeforeEach(func() {
+			var err error
+			kbc, err = utils.NewTestContext(pluginutil.KubebuilderBinName, "GO111MODULE=on")
+			Expect(err).NotTo(HaveOccurred())
+			Expect(kbc.Prepare()).To(Succeed())
+		})
+
+		AfterEach(func() {
+			kbc.Destroy()
+		})
+
+		It("should generate a runnable project with sample external plugin", func() {
+			GenerateProject(kbc)
+		})
+
+	})
+})
+
+// GenerateProject implements a sampleexternalplugin/v1 external plugin project defined by a TestContext.
+func GenerateProject(kbc *utils.TestContext) {
+	var err error
+
+	By("initializing a project")
+	err = kbc.Init(
+		"--plugins", "sampleexternalplugin/v1",
+		"--domain", "sample.domain.com",
+	)
+	ExpectWithOffset(1, err).NotTo(HaveOccurred())
+
+	initFileContainsExpr, err := pluginutil.HasFileContentWith(
+		filepath.Join(kbc.Dir, "initFile.txt"),
+		`A simple text file created with the`+" `"+`init`+"`"+` subcommand
+DOMAIN: sample.domain.com`)
+	ExpectWithOffset(1, err).NotTo(HaveOccurred(), "Check initFile.txt should return no error.")
+	ExpectWithOffset(1, initFileContainsExpr).To(BeTrue(), "The init file does not contain the expected expression.")
+
+	By("creating API definition")
+	err = kbc.CreateAPI(
+		"--plugins", "sampleexternalplugin/v1",
+		"--number=2",
+		"--group", kbc.Group,
+		"--version", kbc.Version,
+		"--kind", kbc.Kind,
+	)
+	ExpectWithOffset(1, err).NotTo(HaveOccurred())
+
+	apiFileContainsExpr, err := pluginutil.HasFileContentWith(
+		filepath.Join(kbc.Dir, "apiFile.txt"),
+		`A simple text file created with the`+" `"+`create api`+"`"+` subcommand
+NUMBER: 2`)
```

Comment:
Ditto on the formatting of the string here. The main goal is to make it as easy as possible to understand what the expected value is without having to do too much mental string manipulation.